test/amoc_api_handler_SUITE.erl

```diff
@@ -0,0 +1,263 @@
+-module(amoc_api_handler_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-export([all/0, init_per_testcase/2, end_per_testcase/2]).
+
+-export([test_success/1,
+         test_fail/1,
+         test_list/1,
+         test_start_fail_no_scenario/1,
+         test_start_success/1,
+         test_load_good/1,
+         test_load_compilation_err/1,
+         test_ping_nodes/1,
+         test_get_test_result/1,
+         test_get_tests_results/1,
+         test_get_wrong_test_result/1]).
+
+-record(state, {action}).
+
+all() ->
+    [test_success,
+        test_fail,
+        test_list,
+        test_start_fail_no_scenario,
+        test_start_success,
+        test_load_good,
+        test_load_compilation_err,
+        test_ping_nodes,
+        test_get_test_result,
+        test_get_tests_results,
+        test_get_wrong_test_result
+        ].
+
+init_per_testcase(test_load_good, Config) ->
+    file:make_dir("scenarios"),
+    file:make_dir("ebin"),
+    Config;
+        
+init_per_testcase(test_load_compilation_err, Config) ->
+    file:make_dir("scenarios"),
+    Config;
+
+init_per_testcase(test_start_success, Config) ->
+    file:make_dir("scenarios"),
+    file:write_file("scenarios/mongoose_simple.erl", get_scenario(), [write]),
+    file:make_dir("ebin"),
+    compile:file("scenarios/mongoose_simple.erl",[{outdir,"ebin"}]),
+    code:purge(mongoose_simple),
+    Config;
+
+init_per_testcase(_, Config) ->
+    application:ensure_all_started(inets),
+    Config.
+
+end_per_testcase(test_load_good, Config) ->
+    file:delete("scenarios/mongoose_simple.erl"),
+    file:del_dir("scenarios"),
+    file:del_dir("ebin"),
+    Config;
+
+end_per_testcase(test_load_compilation_err, Config) ->
+    file:del_dir("scenarios"),
+    Config;
+
+end_per_testcase(test_start_success, Config) ->
+    file:delete("scenarios/mongoose_simple.erl"),
+    file:del_dir("scenarios"),
+    file:delete("ebin/mongoose_simple.beam"),
+    file:del_dir("ebin"),
+    code:purge(mongoose_simple),
+    Config;
+
+end_per_testcase(_, _Config) ->
+    application:stop(inets),
+    application:stop(amoc),
+    amoc_api:stop().
+
+test_success(_Config) ->
```

Comment:
The function names for the test cases have to be more descriptive: "should_confirm_amoc_is_alive" for example.